Code/Framework/AzFramework/AzFramework/Physics/SimulatedBodies/RigidBody.h

```diff
@@ -39,8 +39,8 @@ namespace AzPhysics
 
         virtual void AddShape(AZStd::shared_ptr<Physics::Shape> shape) = 0;
         virtual void RemoveShape(AZStd::shared_ptr<Physics::Shape> shape) = 0;
-        virtual AZ::u32 GetShapeCount() { return 0; }
-        virtual AZStd::shared_ptr<Physics::Shape> GetShape([[maybe_unused]]AZ::u32 index) { return nullptr; }
+        virtual AZ::u32 GetShapeCount() const { return 0; }
+        virtual AZStd::shared_ptr<Physics::Shape> GetShape([[maybe_unused]]AZ::u32 index) const { return nullptr; }
```

Comment:
update: Here we should return `AZStd::shared_ptr<const Physics::Shape>` as this is a `const` member function

Reply:
Added that. 
The only issue is unlike with pointers `shared_ptr<Shape>` can't be assigned to` shared_ptr<const Shape>` since these are different types and that requires to do ugly conversions between them.

Reply:
Hmm I think that should be possible, you can go from non-const to const, but not the other way around (https://godbolt.org/z/bYzdaYzEx)

Reply:
Ok that makes it not a problem at all. 
Just a matter of updating interfaces accepting `const shared_ptr<T>&` when not actually changing the object (like I had to do with DrawHeightfield). thanks!